src/App.tsx

```diff
@@ -15,33 +15,72 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPHA,
+  LENGTH,
+}
+
+export const App: React.FC = () => {
+  const goods = [...goodsFromServer];
+  const [isStarted, switchStart] = useState(false);
+  const [sortBy, setSortBy] = useState(SortType.NONE);
+  const [isReversed, switchReverse] = useState(false);
+  const reset = () => {
+    switchReverse(false);
+    setSortBy(SortType.NONE);
+  };
+
+  goods.sort((g1, g2) => {
+    switch (sortBy) {
+      case SortType.ALPHA:
+        return g1.localeCompare(g2);
+
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      default:
+        return 0;
+    }
+  });
```

Comment:
I would suggest one more option you could have used: just create handlers and pass them to buttons, and when those button are clicked – perform the action (i.e. sort, reset, show – and so on)

why: every time your component gets re-rendered – you perform this action, is this okay?